### PR TITLE
Pub 1080 schedule delete

### PR DESCRIPTION
**Pull request by kurt-uk**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PUB-1080

### Change description ###
added cron scheduler to removal of blobs and artefacts that have a display to date that is before the current date of the check, uses date not datetime and checks 00:00 every day for artefacts yesterday or older

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by commitBlob — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java [hunk 1]
```diff
@@ -253,9 +252,21 @@ public String sendArtefactForSubscription(Artefact artefact) {
...
+
+    /**
+     * Scheduled method that checks daily for newly outdated artefacts based on a yesterday or older display to date
+     */
+    @Scheduled(cron = "${cron.daily-start-of-day}")
+    public void deleteExpiredBlobs() {
+        List<Artefact> outdatedArtefacts = artefactRepository.findOutdatedArtefacts();
+        outdatedArtefacts.forEach(artefact -> log.info(azureBlobService.deleteBlob(artefact.getSourceArtefactId(),
+                                                                   artefact.getProvenance())));
+        artefactRepository.deleteAll(outdatedArtefacts);
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:269]: We should also delete the data from the blob store too

File: src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java [hunk 2]
```diff
@@ -614,4 +615,24 @@ void testDeleteArtefactByIdThrows() {
...
+            assertEquals("1 outdated artefacts found and deleted for before " + LocalDate.now(),
+                         logCaptor.getInfoLogs().get(1), MESSAGES_MATCH);
+        } catch (Exception ex) {
+            throw new IOException(ex.getMessage());
+        }
+    }
+
+    @Test
+    void testDeleteExpiredBlobsWithNoBlobsFound() {
+
```
Comment (ChrisS1512) [inline src/test/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationServiceTest.java:635]: Empty test

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java (hunk 1 above)
Comment (kurt-uk) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:269]: thats what line 267 is doing

Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java:269]: Ah yep, missed that!

**Review by ChrisS1512 — DISMISSED** [review]

**Review by ChrisS1512 — DISMISSED** [review]

**Review by ChrisS1512 — DISMISSED** [review]

**Review by junaidiqbalmoj — CHANGES_REQUESTED** [review]
Can you add the test in which it return the artefact but display_to date is empty. In this case, it should not delete that artefact.

**Review by junaidiqbalmoj — APPROVED** [review]
Tested, SQL statement is working as expected. Not return record with date_to null.

**Review by ChrisS1512 — APPROVED** [review]

**Review by furnivall — APPROVED** [review]
